internal/controller/clusterpolicy_controller.go

```diff
@@ -175,6 +175,26 @@ func templateResourceFilters(kinds []string) kyvernov1.ResourceFilters {
 	return resourceFilters
 }
 
+func (r *ClusterPolicyReconciler) GetClusterPolicy(ctx context.Context, policyName string) (kyvernov1.ClusterPolicy, error) {
+	// Check if the policy is in the cache
+	if policy, ok := r.PolicyCache[policyName]; ok {
+		return policy, nil
+	}
+
+	// If it's not in the cache, query the policy
```

Comment:
Same as the first comment, we shouldn't condition the Get of the ClusterPolicy, that should always happen, and I would personally prefer if that happens in the Reconcile method directly.